zerver/tests/test_settings.py

```diff
@@ -190,6 +199,54 @@ def test_change_email_batching_period(self) -> None:
         hamlet = self.example_user("hamlet")
         self.assertEqual(hamlet.email_notifications_batching_period_seconds, 300)
 
+        # Test that timestamps get updated for existing ScheduledMessageNotificationEmail rows
+        hamlet_msg_id1 = self.send_stream_message(sender=cordelia, stream_name="Verona")
+        hamlet_msg_id2 = self.send_stream_message(sender=cordelia, stream_name="Verona")
+        othello_msg_id1 = self.send_stream_message(sender=cordelia, stream_name="Verona")
+
+        def create_entry(user_profile_id: int, message_id: int, timestamp: datetime) -> int:
+            # The above messages don't actually mention anyone. We just fill up the trigger
+            # because we need to.
+            entry = ScheduledMessageNotificationEmail.objects.create(
+                user_profile_id=user_profile_id,
+                message_id=message_id,
+                trigger=NotificationTriggers.MENTION,
+                scheduled_timestamp=timestamp,
+            )
+            return entry.id
+
+        def get_datetime_object(minutes: int) -> datetime:
+            return datetime(
+                year=2021, month=8, day=10, hour=10, minute=minutes, second=15, tzinfo=timezone.utc
+            )
+
+        hamlet_timestamp = get_datetime_object(10)
+        othello_timestamp = get_datetime_object(20)
+
+        hamlet_entry1_id = create_entry(hamlet.id, hamlet_msg_id1, hamlet_timestamp)
+        hamlet_entry2_id = create_entry(hamlet.id, hamlet_msg_id2, hamlet_timestamp)
+        othello_entry1_id = create_entry(othello.id, othello_msg_id1, othello_timestamp)
+
+        # Update Hamlet's setting from 300 seconds (5 minutes) to 600 seconds (10 minutes)
+        result = self.client_patch(
+            "/json/settings", {"email_notifications_batching_period_seconds": 10 * 60}
```

Comment:
Can you add an assertion that the current batching period was 5*60 before this request?  Will make the test more readable.

Reply:
Done